x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.tsx

```diff
@@ -276,216 +389,149 @@ const AlertsTable: React.FunctionComponent<AlertsTableProps> = (props: AlertsTab
+    return {
       bulkActions,
       alertsCount,
-      rowSelection,
-      alerts: alertsData.alerts,
-      updatedAt,
+      rowSelection: bulkActionsState.rowSelection,
+      alerts,
       isLoading,
       columnIds: visibleColumns,
       onToggleColumn,
       onResetColumns,
       browserFields,
-      controls: props.controls,
+      controls,
       setIsBulkActionsLoading,
       clearSelection,
       refresh,
       fieldBrowserOptions,
       getInspectQuery,
       showInspectButton,
-      toolbarVisibilityProp: props.toolbarVisibility,
-    });
+      toolbarVisibilityProp,
+    };
   }, [
-    bulkActionsState,
     bulkActions,
     alertsCount,
-    alertsData.alerts,
-    updatedAt,
+    bulkActionsState,
     isLoading,
     visibleColumns,
     onToggleColumn,
     onResetColumns,
     browserFields,
-    props.controls,
     setIsBulkActionsLoading,
     clearSelection,
     refresh,
     fieldBrowserOptions,
     getInspectQuery,
     showInspectButton,
-    props.toolbarVisibility,
-  ])();
+    toolbarVisibilityProp,
+    alerts,
+    controls,
+  ]);
 
-  const leadingControlColumns = useMemo(() => {
-    let controlColumns = [...props.leadingControlColumns];
+  const toolbarVisibility = useGetToolbarVisibility(toolbarVisibilityArgs);
 
-    if (CustomActionsRow) {
-      controlColumns = [
-        {
+  const customActionsRow = useMemo(() => {
+    return renderCustomActionsRow
+      ? {
           id: 'expandColumn',
           width: actionsColumnWidth,
-          headerCellRender: () => {
-            return (
-              <span data-test-subj="expandColumnHeaderLabel">
-                {ALERTS_TABLE_CONTROL_COLUMNS_ACTIONS_LABEL}
-              </span>
-            );
-          },
-          rowCellRender: (cveProps: EuiDataGridCellValueElementProps) => {
-            const { visibleRowIndex } = cveProps as EuiDataGridCellValueElementProps & {
-              visibleRowIndex: number;
-            };
-
-            if (!ecsAlertsData[visibleRowIndex]) {
-              return null;
-            }
-
-            return (
-              <EuiFlexGroup gutterSize="none" responsive={false} alignItems="center">
-                <CustomActionsRow
-                  id={props.id}
-                  alert={alerts[visibleRowIndex]}
-                  ecsAlert={ecsAlertsData[visibleRowIndex]}
-                  nonEcsData={oldAlertsData[visibleRowIndex]}
-                  rowIndex={visibleRowIndex}
-                  setFlyoutAlert={handleFlyoutAlert}
-                  setIsActionLoading={getSetIsActionLoadingCallback(visibleRowIndex)}
-                  cveProps={cveProps}
-                  refresh={refresh}
-                  clearSelection={clearSelection}
-                />
-              </EuiFlexGroup>
-            );
-          },
-        },
-        ...controlColumns,
-      ];
-    }
+          headerCellRender: ControlColumnHeaderRenderCell,
+          rowCellRender: ControlColumnRowRenderCell,
```

Comment:
@kqualters-elastic here the table `id` is not being passed to the custom actions cell renderer. In order for the `View rule details` and `View alert details` actions to work correctly that id is needed:

```diff
const AlertsTable: React.FunctionComponent<AlertsTableProps> = memo((props: AlertsTableProps) => {
   const {
+    id,
     visibleColumns,
     onToggleColumn,
     onResetColumns,

...

           id: 'customActions',
           width: actionsColumnWidth,
           headerCellRender: ControlColumnHeaderRenderCell,
-          rowCellRender: ControlColumnRowRenderCell,
+          rowCellRender: (cellProps) => <ControlColumnRowRenderCell {...cellProps} id={id} />,
         }
       : undefined;
   }, [renderCustomActionsRow, actionsColumnWidth]);

```

Reply:
fixed in https://github.com/elastic/kibana/pull/180016/commits/dc425e2f44850b6115ec3c4c9abf92ef2373ee8a